src/clusterfuzz/_internal/system/archive.py

```diff
@@ -407,6 +412,103 @@ class ArchiveType:
   TAR_LZMA = 3
 
 
+@dataclasses.dataclass
+class CacheBlock:
+  """Represents a cache entry for the HttpZipFile."""
+  start: int
+  end: int
+  content: Sequence[bytes]
+
+
+class HttpZipFile:
+  """This class is a very simple file-object representation of a file over HTTP.
+  It uses the 'Accept-Ranges' feature of HTTP to fetch parts (or all) of the
+  file.
+  """
+
+  @staticmethod
+  def is_uri_compatible(uri: str) -> bool:
+    try:
+      res = requests.head(uri, timeout=5)
+      return res.headers.get('Accept-Ranges') is not None
+    except:
+      return False
+
+  def __init__(self, uri):
+    self.uri = uri
+    self.session = requests.Session()
+    # This slows down the overall performances.
+    self.session.headers.pop('Accept-Encoding')
+    resp = self.session.head(self.uri)
+    self.file_size = int(resp.headers.get('Content-Length', default=0))
+    self._current_block = CacheBlock(0, 0, [])
+    self._pos = 0
+    assert resp.headers.get('Accept-Ranges') is not None
```

Comment:
I honestly do not think this should be a concern. Tbh, nothing prevents that from happening after we created the object, and we cannot really prevent that as well (and we'll end up in a very uncomfortable situation as well). Should we really try to prevent from that? (that's an honest question tbf)